### ISSUE TITLE
AttributeError: 'dict' object has no attribute 'authorize' error after having moved to google.oauth2.credentials.Credentials

### ISSUE THREAD

**Issue by andreamoro** [googleapis/google-auth-library-python#1400]
#### Environment details

  - OS: Ubuntu 23
  - Python version: 3.12
  - pip version: 23.3.1
  - google-api-python-client: 2.105.0
  - google-auth: 2.23.3
  - google-auth-httplib2: 0.1.1
  - google-auth-oauthlib: 1.1.0

#### Steps to reproduce
I've been trying to upgrade my code from the deprecated `oauth2client.client.OAuth2Credentials` to `google.oauth2.credentials.Credentials` without success as I continue to get the authorize attribute error being missing despite having upgraded all the librarie.
Note: Although I created a brand new environment, after the first pip install I had to upgrade some libraries as there were not aligned with the above details.

That said, after having obtained a token with a flow object, I then get a credential object via the `google.oauth2.credentials.Credentials` (successfully)

![image](https://github.com/googleapis/google-auth-library-python/assets/5256583/03efe6a6-5800-432f-9c72-11054ad167ec)

However, by the time I pass the credentials along with the API version and name to the discovery.build method, I get the following 
AttributeError message. 

![image](https://github.com/googleapis/google-auth-library-python/assets/5256583/fc86d613-635f-4fab-9b02-74d87e9d7acf)

See below the incriminated code:
```
        API_SERVICE_NAME = 'searchconsole'
        API_VERSION = 'v1'

        search_console_service = discovery.build(
            API_SERVICE_NAME, API_VERSION, credentials=credentials)
```
and credentials dictionary
```
    'token': 'redacted'
    'refresh_token': 'redacted'
    'client_id': 'redacted'
    'client_secret': 'redacted'
    'token_uri': 'https://accounts.google.com/o/oauth2/token'
    'scopes': 'https://www.googleapis.com/auth/webmasters.readonly'
```
Any idea of what's wrong?
Thanks!

**Comment by clundin25** [comment]
Hey can you share a sample that reproduces this issue? It is hard to follow the flow in this query.

**Comment by andreamoro** [comment]
Actually, ignore me. I just realised even from the small things above, that I was no longer passing the authenticated credential objects to the build method but my dictionary.
Apologies for the confusion.
